validator/main.py

```diff
@@ -115,43 +120,48 @@ def get_args(self) -> Dict[str, Any]:
             "model_threshold": self._kwargs.get("model_threshold", 0.5),
         }
 
-    def get_topics_ensemble(self, text: str, candidate_topics: List[str]) -> List[str]:
+    def get_topics_ensemble(self, text: str, candidate_topics: List[str]):
         applicable_topics = []
 
-        for candidate_topic in candidate_topics:
-            candidates = [candidate_topic, "other"]
-            topic, confidence = self.get_topic_zero_shot(text, candidates)
-
-            if confidence < self._model_threshold:
+        topics, scores = self.get_topic_zero_shot(text, candidate_topics)
+        for topic, score in zip(topics, scores):
+            if score > self._model_threshold:
+                applicable_topics.append(topic)
+            elif score < self._model_threshold:
                 response = self.call_llm(text, candidate_topics)
                 topic = json.loads(response)["topic"]
-
-            if topic != "other":
+            if topic in self._invalid_topics:
```

Comment:
Do we actually need this conditional? Since we're iterating over `candidate_topics`, and the only time this function is called `candidate_topics` is set as `self._invalid_topics`, by definition topic will always belong to `self._invalid_topics`.